src/main/java/org/hydev/mcpm/server/crawlers/spiget/CreateDatabase.java

```diff
@@ -0,0 +1,174 @@
+package org.hydev.mcpm.server.crawlers.spiget;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.hc.client5.http.utils.Hex;
+import org.hydev.mcpm.client.models.Database;
+import org.hydev.mcpm.client.models.PluginModel;
+import org.hydev.mcpm.client.models.PluginVersion;
+import org.hydev.mcpm.client.models.PluginYml;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.yaml.snakeyaml.error.YAMLException;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Responsible for statically generating database "index" files for mcpm.
+ * Used for searches, etc. To create a database use the createDatabase method
+ * with a path to the .mcpm/pkgs/spiget directory generated previously by SpigetCrawler.
+ */
+public class CreateDatabase {
+    public static final String packageStore = ".mcpm";
+
+    /**
+     * Main method generates a database in the mcpm directory.
+     *
+     * @param args Arguments are ignored.
+     */
+    public static void main(String[] args) {
+        writeDatabase(new File(packageStore, "pkgs/spiget"), new File(packageStore, "db"));
+    }
+
+    /**
+     * Iterates through the crawlerDirectory and assembles and writes a database object to databaseFile.
+     *
+     * @param crawlerDirectory Path to the .mcpm/pkgs/spiget directory generated by SpigetCrawler.
+     * @param databaseFile Path to the db.json file that will be written to.
+     */
+    public static void writeDatabase(File crawlerDirectory, File databaseFile) {
+        Database database = createDatabase(crawlerDirectory);
+
+        if (database == null)
+            return;
+
+        try {
+            new ObjectMapper().writeValue(databaseFile, database);
```

Comment:
It would be nice if you can implement zstd compression here as well. You can try https://github.com/luben/zstd-jni

Reply:
Yeah I agree. Do you want me to do this In-PR? I'm tempted to push this off to when DatabaseInteractor has some meat to it (mostly because I'm lazy TBH).

Reply:
Okay yea that's reasonable